File: Java Script - Introduction to Programming/4.1. Complex Conditions/13. Point in the Figure.js
function pointInFigure([...arg]) {
    let [h, x, y] = [...arg.map(Number)]

    if (x > 0 && x < 3 * h && y > 0 && y < h || x > h && x < 2 * h && y > 0 && y < 4 * h) {
        return 'inside'
    } else if (x == 0 && y >= 0 && y <= h ||
        x == 3 * h && y >= 0 && y <= h ||
        y == 0 && x >= 0 && x <= 3 * h ||
        y == h && x >= 0 && x <= 3 * h ||
        x == h && y >= 0 && y <= 4 * h ||
        x == 2 * h && y >= 0 && y <= 4 * h ||
        y == 0 && x >= h && x <= 2 * h ||
        y == 4 * h && x >= h && x <= 2 * h) {
        return 'border'
    } else {
        return 'outside'
    }
}

// console.log(pointInFigure('leap', 5, 2))

File: Pyrhon - Introduction to Programming/4.1. Complex Conditions/13. Point in the Figure.py
h, x, y = [float(input()) for _ in range(3)]

if 0 < x < 3 * h and 0 < y < h or h < x < 2 * h and 0 < y < 4 * h:
    print('inside')
elif x == 0 and 0 <= y <= h or \
        x == 3 * h and 0 <= y <= h or \
        y == 0 and 0 <= x <= 3 * h or \
        y == h and 0 <= x <= 3 * h or \
        x == h and 0 <= y <= 4 * h or \
        x == 2 * h and 0 <= y <= 4 * h or \
        y == 0 and h <= x <= 2 * h or \
        y == 4 * h and h <= x <= 2 * h:
    print('border')
else:
    print('outside')


'''
2
3
10
#outside
'''
